financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -91,19 +91,21 @@ class ExpiringMultiPartyClient {
   }
 
   async update() {
-    this.collateralRequirement = this.toBN((await this.emp.methods.collateralRequirement().call()).toString());
-    this.liquidationLiveness = Number(await this.emp.methods.liquidationLiveness().call());
-
-    const events = await this.emp.getPastEvents("NewSponsor", { fromBlock: 0 });
+    const [collateralRequirement, liquidationLiveness, events, cumulativeFeeMultiplier] = await Promise.all([
```

Comment:
All of the contract calls should be grouped into as few Promise.all's as possible.

Reply:
Good, refactor here. I'm actually working on this kind of PR right now in decreasing the number of calls the bots make. Calling `this.emp.methods.collateralRequirement().call()`, `this.emp.methods.liquidationLiveness().call()` & `this.emp.methods.cumulativeFeeMultiplier().call()` actually only needs to happen _once_ when the bot boots, not every loop. Moving this into a separate update function that only gets called at the start will help a bunch with this.

Reply:
+1. I also couldn't figure out the best way to combine the `positions` Promise array in the Promise.all but once this PR is merged you should rebase and try to add that + `getCurrentTime()` to one single Promise.all!

Reply:
@chrismaree update: I just pushed a commit that calls `positions()` and `getLiquidations()` in parallel. This cuts the run time by 50%

Reply:
ye, this looks stellar. I've come up with a few other improvements in PR [1813](https://github.com/UMAprotocol/protocol/pull/1813) that are similar in other parts of the codebase. Let's get this one merged in for now and I'll add on top.